src/main/java/de/hpi/bpt/argos/api/eventQuery/EventQueryEndpointImpl.java

```diff
@@ -0,0 +1,219 @@
+        sparkService.delete(EventQueryEndpoint.getDeleteEventQueryBaseUri(), this::deleteEventQuery);
+        sparkService.put(EventQueryEndpoint.getEditEventQueryBaseUri(), this::editEventQuery);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        response.body("");
+
+        try {
+            JsonObject jsonEventQuery = jsonParser.parse(request.body()).getAsJsonObject();
+
+            if (jsonEventQuery == null) {
+                halt(HttpStatusCodes.BAD_REQUEST, "no event query given in body");
+            }
+
+            EventQuery eventQuery = getEventQueryFromJson(jsonEventQuery, true);
+            if (eventQuery == null) {
+                halt(HttpStatusCodes.BAD_REQUEST, "cannot parse event query");
+            }
+
+            checkEventTypeExists(eventQuery);
+
+            EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().registerEventQuery(eventQuery.getTypeId(), eventQuery);
+
+            if (!feedback.isSuccessful()) {
+                halt(HttpStatusCodes.ERROR, String.format("cannot register event type: %1$s", feedback.getResponseText()));
+            }
+
+            PersistenceAdapterImpl.getInstance().createArtifact(eventQuery, EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()));
+
+        } catch (HaltException halt) {
+            LoggerUtilImpl.getInstance().error(logger,
+                    String.format("cannot create event query: %1$s -> %2$s", halt.statusCode(), halt.body()), halt);
+            throw halt;
+        } catch (Exception e) {
+            LoggerUtilImpl.getInstance().error(logger, JSON_PARSE_ERROR_MESSAGE, e);
+            halt(HttpStatusCodes.ERROR, e.getMessage());
+        }
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+		response.body("");
+
+        long eventQueryId = getEventQueryId(request);
+        EventQuery eventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(eventQueryId);
+        if (eventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "given query was not found");
+        }
+
+        EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().deleteEventQuery(eventQuery);
+        if (!feedback.isSuccessful()) {
+            halt(HttpStatusCodes.ERROR, String.format("cannot unregister event query: %1$s", feedback.getResponseText()));
+        }
+
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifact(eventQuery,
+                EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()))) {

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'eventQuery' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)